runtime/registry.go

```diff
@@ -187,15 +191,17 @@ func (r *Runtime) EditInstance(ctx context.Context, inst *drivers.Instance) erro
 // TODO :: this is a rudimentary solution and ideally should be done in some producer/consumer pattern
 func (r *Runtime) evictCaches(ctx context.Context, inst *drivers.Instance) {
 	// evict and close exisiting connection
-	r.connCache.evict(ctx, inst.ID, inst.OLAPDriver, inst.OLAPDSN)
-	r.connCache.evict(ctx, inst.ID, inst.RepoDriver, inst.RepoDSN)
+	c, _, _ := r.opts.OLAPDef(inst.OLAPDSN)
+	r.connCache.evict(ctx, inst.ID, inst.OLAPDriver, variables(_olapDriverName, c.Defaults, inst.ResolveVariables()))
+	c, _, _ = r.opts.RepoDef(inst.RepoDSN)
+	r.connCache.evict(ctx, inst.ID, inst.RepoDriver, variables(_repoDriverName, c.Defaults, inst.ResolveVariables()))
```

Comment:
We can keep `inst.OLAPDSN` and `inst.RepoDSN` for backwards compatibility, but ideally that would just be handled in the APIs.

Elsewhere, could we have `inst.OLAP` and `inst.Repo` be names of connectors, and then just have `inst.Connectors` contain all the declared connectors for it?